common/nym-kcp/src/session.rs

```diff
@@ -0,0 +1,1778 @@
+            xmit: 0,
+            data,
+        }
+    }
+}
+
+impl Default for KcpSession {
+    fn default() -> Self {
+        KcpSession {
+            conv: 0,
+            snd_nxt: 0,
+            snd_una: 0,
+            snd_wnd: 32,
+            rmt_wnd: 128,
+            snd_queue: VecDeque::new(),
+            snd_buf: VecDeque::new(),
+
+            rcv_nxt: 0,
+            rcv_wnd: 128,
+            rcv_buf: VecDeque::new(),
+            rcv_queue: VecDeque::new(),
+
+            rx_srtt: 0,
+            rx_rttval: 0,
+            rx_rto: 3000,
+            rx_minrto: 3000,
+
+            current: 0,
+            interval: 100,
+            ts_flush: 100,
+
+            out_pkts: Vec::new(),
+            mtu: 1376,
+            partial_read: None,
+        }
+    }
+}
+
+impl KcpSession {
+    pub fn ts_current(&self) -> u32 {
+        self.current
+    }
+
+    pub fn ts_flush(&self) -> u32 {
+        self.ts_flush
+    }
+
+    fn available_send_segments(&self) -> usize {
+        // A naive approach: if `snd_queue` has length L
+        // and local window is `snd_wnd`, we can add `snd_wnd - L` more segments
+        let used = self.snd_queue.len();
+        let allowed = self.snd_wnd as usize;
+        allowed.saturating_sub(used)
+    }
+
+    /// Create a new KCP session with a specified conv ID and default MSS.
+    pub fn new(conv: u32) -> Self {
+        KcpSession {
+            conv,
+            ..Default::default()
+        }
+    }
+
+    /// If you want to let the user set the mtu:
+    pub fn set_mtu(&mut self, mtu: usize) {
+        self.mtu = mtu;
+    }
+
+    /// Set the update interval (flush interval) in milliseconds
+    pub fn set_interval(&mut self, interval: u32) {
+        let interval = interval.clamp(10, 5000);
+        self.interval = interval;
+    }
+
+    /// Manually set the minimal RTO
+    pub fn set_min_rto(&mut self, rto: u32) {
+        self.rx_minrto = rto;
+    }
+
+    pub fn send(&mut self, mut data: &[u8]) {
+        debug!("Sending data, len: {}", data.len());
+
+        if data.is_empty() {
+            return;
+        }
+
+        // How many segments do we need?
+        // If data <= mss, it's 1; otherwise multiple.
+        let total_len = data.len();
+        let count = if total_len <= self.mtu {
+            1
+        } else {
+            total_len.div_ceil(self.mtu)
+        };
+
+        debug!("Will send {} fragments", count);
+
+        // Build each fragment
+        for i in 0..count {
+            let size = std::cmp::min(self.mtu, data.len());
+            let chunk = &data[..size];
+
+            // KCP fragment numbering is REVERSED - last fragment has frg=0,
+            // first has frg=count-1. This allows receiver to know total count from first packet.
+            // In KCP, `frg` is set to the remaining fragments in reverse order.
+            // i.e., the last fragment has frg=0, the first has frg=count-1.
+            let frg = (count - i - 1) as u8;
```

Comment:
what if we have a huge data chunk to send, such that `count >= 256`, i.e. the `u8` cast fails?